src/lib/statistics/statistics_objects/abstract_histogram.cpp

```diff
@@ -0,0 +1,1035 @@
+          }
+        }
+        return true;
+      }
+
+      return false;
+    }
+    case PredicateCondition::Like:
+    case PredicateCondition::NotLike:
+      Fail("Predicate (NOT) LIKE is not supported for non-string columns.");
+    default:
+      // Do not prune predicates we cannot (yet) handle.
+      return false;
+  }
+}
+
+template <typename T>
+bool AbstractHistogram<T>::_does_not_contain(const PredicateCondition predicate_condition,
+                                             const AllTypeVariant& variant_value,
+                                             const std::optional<AllTypeVariant>& variant_value2) const {
+  return _general_does_not_contain(predicate_condition, variant_value, variant_value2);
+}
+
+template <>
+bool AbstractHistogram<pmr_string>::_does_not_contain(const PredicateCondition predicate_condition,
+                                                      const AllTypeVariant& variant_value,
+                                                      const std::optional<AllTypeVariant>& variant_value2) const {
+  const auto value = static_variant_cast<pmr_string>(variant_value);
+  if (!value) {
+    return false;
+  }
+
+  switch (predicate_condition) {
+    case PredicateCondition::Like: {
+      if (!LikeMatcher::contains_wildcard(*value)) {
+        return _does_not_contain(PredicateCondition::Equals, *value);
+      }
+
+      // If the pattern starts with a MatchAll, we can not prune it.
+      if (value->front() == '%') {
+        return false;
+      }
+
+      /**
+       * We can prune prefix searches iff the domain of values captured by a prefix pattern is prunable.
+       *
+       * Example:
+       * bins: [a, b], [d, e]
+       * predicate: col LIKE 'c%'
+       *
+       * With the same argument we can also prune predicates in the form of 'c%foo',
+       * where foo can be any pattern itself.
+       * We only have to consider the pattern up to the first AnyChars wildcard.
+       */
+      const auto match_all_index = value->find('%');
+      if (match_all_index != pmr_string::npos) {
+        const auto search_prefix = value->substr(0, std::min(match_all_index, _domain.prefix_length));
+        if (_does_not_contain(PredicateCondition::GreaterThanEquals, search_prefix)) {
+          return true;
+        }
+
+        const auto search_prefix_next_value =
+            StringHistogramDomain{_domain.min_char, _domain.max_char, search_prefix.length()}.next_value(search_prefix);
+
+        // If the next value is the same as the prefix, it means that there is no larger value in the domain
+        // of substrings. In that case we cannot prune, because otherwise the previous check would already return true.
+        if (search_prefix == search_prefix_next_value) {
+          return false;
+        }
+
+        if (_does_not_contain(PredicateCondition::LessThan, search_prefix_next_value)) {
+          return true;
+        }
+
+        const auto search_prefix_bin = _bin_for_value(search_prefix);
```

Comment:
By now you have checked if there is no value greater than our prefix and no value smaller than the next value. I thought you'd be done by now. Please briefly explain what happens next.

Reply:
"you" is the one addressee here, this is Tim's doing ;). in person.  let's make this dependent on whether we keep like estimations in the histograms.

Reply:
killed.